docs/proposals/mqtt.md

```diff
@@ -0,0 +1,234 @@
+   standard MQTT Library APIs like [Eclipse Paho MQTT Library](https://www.eclipse.org/paho/index.php?page=downloads.php).
+
+    a. (OPTIONAL - Security) Username and Password can be provided in the MQTT Library APIs to provide security. If TLS Security is not implemented the 
+       Username/Password is sent as plain-text
+    
+    b. (OPTIONAL - Last Will and Testament) Last Will and Testament concept helps other MQTT Clients know if the a client has been disconnected abruptly.
+       This might be an interesting concept if the Monitoring Broker is able to send information back to the IoT Devices as a way to control them.  A concise
+       description of MQTT LW&T Concept can be found [here](https://www.hivemq.com/blog/mqtt-essentials-part-9-last-will-and-testament/)
+
+      > NOTE: In a well developed MQTT Application, IoT Devices should provide a Will message which the MQTT Monitoring Broker can use to detect if these
+         devices are still available and their related applications should be stopped or kept-running
+
+2. After determining if the MQTT CONNECT message is not malformed, the MQTT Broker Server sends an Acknowledgement back the Monitoring Broker with 
+   MQTT CONNACK command packet.
+
+3. Upon successful connection with the MQTT Broker Server, the Monitoring Broker can subscribe to a list of MQTT topics with possible Quality of Service
+   (QoS) parameter either set to `1` or `2` depending on the guarantees required (generally, a QoS of 1 should work)
+
+4. The Broker Server then sends back an Acknowledgement of the requested subscriptions via the SUBACK packet. This packet contains return codes for all the
+   list of topics and their QoS. 
+
+5. If the IoT Devices already have been publishing data prior to connection of Monitoring Broker with an assumption that they publish their data with QoS 1,
+   the Broker Server would publish all the information to the Monitoring Broker (persistent storage). If IoT Devices are yet to publish on their dedicated
+   topics, the MQTT Broker Server will publish data to the Monitoring Broker once the data is available.
+
+A possible integration of Akri's MQTT Monitoring Broker and MQTT Broker Server looks as follows:
+
+![Akri MQTT Monitoring Broker and MQTT Broker Server System Diagram](../media/akri_mqtt_broker_server.png)
+
+
+## Questions
+
+- Would a generic MQTT Monitoring Broker be possible to implement with Akri?
+    - If not, maybe would it be wise to begin with a Specification like [Sparkplug MQTT Specification](https://sparkplug.eclipse.org) where the Topic Structure
+      and the Payload are defined?
+
+- Is gRPC Server required within the MQTT Monitoring Broker? Does the Publish/Subscribe mechanism require a different technology?
+
+- How actually would Discovery Handler Look like? When one talks about Discovery Handler, is this pertaining to only MQTT Broker Server or the discovery of leaf
```

Comment:
It seems like this should have its own section in the proposal with the following question also seeming to fit in a "What does discovery mean?" section. From previous discussions, it seems like Akri's role is discovering MQTT Broker Servers.